namer/consul/src/main/scala/io/buoyant/namer/consul/LookupCache.scala

```diff
@@ -41,13 +42,21 @@ private[consul] class LookupCache(
     }
   }
 
-  // Note that this activity is never recomputed.  We simply do a
+  private[this] def resolveDc(Datacenter: String): Activity[String] = localDc match {
+    case Some(Datacenter) => config.map(_.flatMap(_.Datacenter).getOrElse(Datacenter))
+    case _ => Activity.value(Datacenter)
+  }
+
+  // Note that this 2 activities are never recomputed.  We simply do a
   // lookup for the domain and then wrap it an activity for
   // convenience.
+  private[this] lazy val config: Activity[Option[v1.Config]] =
```

Comment:
I thought about that but wasn't sure where we need an extra activity just for that... But if you say that it will clarify things - then sure, will do!